### ISSUE TITLE
Using LinqToDb.Sql.Property<string> in a called method resolves predicate as System.Object and not the data model

### ISSUE THREAD

**Issue by th3hunter** [linq2db/linq2db#2817]
I'm trying to create a method that dynamically builds filters for generic CRUDs, based on a parameter list passed in a JSON object.

What am I doing wrong? Maybe this is not fully implemented yet?

```
Exception message:
'Property 'System.String Nombre' is not defined for type 'System.Object''

Stack trace:
Se produjo una excepción de tipo 'System.ArgumentException' en linq2db.dll pero no se controló en el código del usuario: 
   en System.Linq.Expressions.Expression.Property(Expression expression, PropertyInfo property)
   en System.Linq.Expressions.Expression.MakeMemberAccess(Expression expression, MemberInfo member)
   en LinqToDB.Linq.Builder.ExpressionBuilder.<ConvertExpression>b__119_0(Expression e)
   en LinqToDB.Expressions.Extensions.Transform(Expression expr, Func`2 func)
   en LinqToDB.Expressions.Extensions.Transform(Expression expr, Func`2 func)
   en LinqToDB.Linq.Builder.ExpressionBuilder.ConvertExpression(Expression expression)
   en LinqToDB.Linq.Builder.ExpressionBuilder.BuildWhere(IBuildContext parent, IBuildContext sequence, LambdaExpression condition, Boolean checkForSubQuery, Boolean enforceHaving)
   en LinqToDB.Linq.Builder.WhereBuilder.BuildMethodCall(ExpressionBuilder builder, MethodCallExpression methodCall, BuildInfo buildInfo)
   en LinqToDB.Linq.Builder.MethodCallBuilder.BuildSequence(ExpressionBuilder builder, BuildInfo buildInfo)
   en LinqToDB.Linq.Builder.ExpressionBuilder.BuildSequence(BuildInfo buildInfo)
   en LinqToDB.Linq.Builder.ExpressionBuilder.Build[T]()
   en LinqToDB.Linq.Query`1.CreateQuery(IDataContext dataContext, Expression expr)
   en LinqToDB.Linq.Query`1.GetQuery(IDataContext dataContext, Expression& expr)
   en LinqToDB.Linq.ExpressionQuery`1.GetQuery(Expression& expression, Boolean cache)
   en LinqToDB.Linq.ExpressionQuery`1.get_SqlText()
   en LinqToDB.Linq.ExpressionQueryImpl`1.ToString()
   en Servos.Data.QueryHelpers`1.Filter(IQueryable`1 q, Request req) en D:\netcore\Servos\data\QueryHelpers.cs: línea 34
   en Servos.Masters.Usuarios.List(Request req) en D:\netcore\Servos\masters\Usuarios.cs: línea 48
   en Servos.API.Usuarios.List(Request req) en D:\netcore\Servos\API\Controllers\Masters\Usuarios.cs: línea 24
   en Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.SyncObjectResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   en Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeActionMethodAsync()
   en Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   en Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.InvokeNextActionFilterAsync()
```

### Steps to reproduce

**Data model**
```c#
namespace Data
{
        [Table(Schema="public", Name="usuarios")]
	public partial class Usuarios
	{
		[PrimaryKey, NotNull    ] public decimal UsuarioId   { get; set; } // numeric(10,0)
		[Column,     NotNull    ] public string  Nombre      { get; set; } // character varying(60)
		[Column,        Nullable] public string  Password    { get; set; } // character varying(255)
		[Column,     NotNull    ] public char    Estado      { get; set; } // character(1)
	}
}
```
**Parameter class**
```c#
[Serializable]
public class Parameter
{
    public string Name { get; set; }
    public string Value { get; set; }
    public DataType DataType { get; set; } = DataType.STRING;
    public Operator Operator { get; set; } = Operator.EQUAL;
    public bool IncludeInAutomaticFilter { get; set; } = true;
}
```
**Request class**
```c#
    [Serializable]
    public class Request
    {
        public List<Parameter> Parameters { get; set; }
        public dynamic Data { get; set; }
    }
```
**Builder method**
```c#
    public static class QueryHelpers<T>
    {
        public static IQueryable<T> Filter(IQueryable<T> q, Request req)
        {
            var r = q;

            // Iterates the parameter list
            foreach (var param in req.Parameters)
            {
                // STRING
                if (param.DataType == DataType.STRING)
                {
                    switch (param.Operator)
                    {
                        case Operator.EQUAL:
                            r = r.Where(p => Sql.Property<string>(p, param.Name) == param.Value);
                            break;
                        }
                        // .... and so on
                    }
                }
                // .... and so on
            }
            return r;
        }
    }
```
**Usage**
```c#
    public List<Data.Usuarios> GetList(Request req)
    {
        IQueryable<Data.Usuarios> q = from p in db.Usuarios
            orderby p.Nombre
            select p;

        // Build the filters
        q = QueryHelpers<Data.Usuarios>.Filter(q, req);
    }
```
**JSON Request object**
```json
    {
      "parameters": [
        {
          "Name": "Nombre",
          "Value": "ministra",
          "Operator": 0, // EQUAL
          "DataType": 0  // STRING
        }
      ],
      "data": {}
    }
```

### Environment details
linq2db version: *3.2.3*
Database Server: *Postgres 12*
Operating system: *Windows 10*
.NET Framework: *5.0.101*

**Comment by sdanyliv** [comment]
What is definition of `Data.Usuarios` ?

**Comment by th3hunter** [comment]
> What is definition of `Data.Usuarios` ?

Updated and included in "Steps to reproduce".

**Comment by sdanyliv** [comment]
Looks like a bug. I need to check. But I can propose better workaround:
```cs
public static class QueryHelpers
{
    public static IQueryable<T> Filter<T>(this IQueryable<T> query, Request req)
    {
        var entityParam = Expression.Paramater(typeof(T), "e");
        Expression predicate = null;

        // Iterates the parameter list
        foreach (var param in req.Parameters)
        {
            Expression condition = null;
            var property = Expression.PropertyOrField(entityParam, param.Name);
            Expression value = Expression.Constant(param.Value);
            
            // TODO: probably here you need better conversion
            If (property.Type != value.Type)
                value = Expression.Convert(value, property.Type);

            switch (param.Operator)
            {
                case Operator.EQUAL:
                    condition = Expression.Equal(property, value);
                    break;
                }
                // .... and so on
            }
        
            // .... and so on

            if (condtion != null)
            {
                if (predicate == null)
                    predicate = condition;
                else
                    predicate = Expression.AndAlso(predicate, condition);
            }
            
        }

        if (predicate == null)
            predicate = Expression.Constant(false);

        var lambda = Expression.Lambda<Func<T, bool>>(predicate, param);

        return query = query.Where(lambda);
    }
}
```

**Comment by th3hunter** [comment]
This works like a charm, awesome!

**Comment by MaceWindu** [comment]
Fixed in `6.0.0-preview.1`
